cli/enginecli/test_dropper.py

```diff
@@ -1137,36 +1151,106 @@ def test_claim_erc721_with_custom_amount(self):
         )
 
         current_block = len(chain)
-        block_deadline = current_block  # since blocks are 0-indexed
+        block_deadline_0 = current_block  # since blocks are 0-indexed
 
-        request_id = 89390589485
-        message_hash = self.dropper.claim_message_hash(
-            claim_id, request_id, accounts[1].address, block_deadline, custom_amount
-        )
-        signed_message = sign_message(message_hash, self.signer_0)
+        request_id_0 = 89390589485
 
-        self.assertEqual(self.nft_contract.owner_of(token_id), self.dropper.address)
-        self.assertFalse(self.dropper.claim_status(claim_id, request_id))
+        for idx, token_id_as_ammount in enumerate(token_ids):
+            request_id = request_id_0 + idx
+            block_deadline = block_deadline_0 + idx
+            message_hash = self.dropper.claim_message_hash(
+                claim_id,
+                request_id,
+                accounts[1].address,
+                block_deadline,
+                token_id_as_ammount,
+            )
+            signed_message = sign_message(message_hash, self.signer_0)
 
-        self.dropper.claim(
-            claim_id,
-            request_id,
-            block_deadline,
-            custom_amount,
-            self.signer_0,
-            signed_message,
-            {"from": accounts[1]},
+            self.assertEqual(
+                self.nft_contract.owner_of(token_id_as_ammount), self.dropper.address
+            )
+            self.assertFalse(self.dropper.claim_status(claim_id, request_id))
+
+            self.dropper.claim(
+                claim_id,
+                request_id,
+                block_deadline,
+                token_id_as_ammount,
+                self.signer_0,
+                signed_message,
+                {"from": accounts[1]},
+            )
+            self.assertEqual(
+                self.nft_contract.owner_of(token_id_as_ammount), accounts[1].address
+            )
+            self.assertTrue(self.dropper.claim_status(claim_id, request_id))
+
+    def test_claim_erc721_multiple_claimants(self):
```

Comment:
Two more tests:
1. If there is a tokenId 0 on the contract and the drop was created with `tokenId = 0`, then transferring a non-zero token ID will result in the correct token being transferred to the claimant.
2. If the Dropper contract does not own the token ID that's being transferred, the transaction is reverted. This should be covered by `safeTransferFrom` anyway, but good to have an explicit test.
3. If the recipient is a smart contract that does not implement `onERC721Received`, then the transaction is reverted. This should be covered by `safeTransferFrom` as well, but we need a test here as it is part of our interface. You can test this by making a contract with a `claim` method that proxies the claim to the Dropper contract.